### PR TITLE
A4A: Update the Referrals page based on the new design.

### PR DESCRIPTION
**Pull request by jkguidaven**
This PR updates the Referrals page based on the new design.

**Updated Referrals page**
| Before | After |
|--------|--------|
| Screenshot 2024-05-07 at 12 14 11 AM | Screenshot 2024-05-07 at 6 38 11 PM |

**Dedicated Commission details and terms page**
Screenshot 2024-05-07 at 12 11 02 AM

Closes https://github.com/Automattic/automattic-for-agencies-dev/issues/438

## Proposed Changes

* Update the Referrals page with a more simplified text.
* Add a dedicated Commission details and terms page.

## Testing Instructions

* Use the A4A live link and go to `/referrals`
* Confirm that the page matches the new design.
* Click the 'How can I earn?' link.
* Confirm that you are redirected to the new dedicated Commission details and terms page.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/a8c-for-agencies/sections/referrals/primary/footer/index.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { useTranslate } from 'i18n-calypso';
+
+import './style.scss';
+
+export default function ReferralsFooter() {
+	const translate = useTranslate();
+
+	const tosLink = ''; // FIXME: Add the correct link to the Terms of Service.
```
Comment (jkguidaven) [inline client/a8c-for-agencies/sections/referrals/primary/footer/index.tsx]: @madebynoam, I'd like to clarify which TOS link I should use in the footer.

Comment (rcanepa) [reply client/a8c-for-agencies/sections/referrals/primary/footer/index.tsx]: @jkguidaven, the TOS link should point to https://automattic.com/for-agencies/platform-agreement/.

Comment (rcanepa) [inline client/a8c-for-agencies/sections/referrals/primary/footer/index.tsx]:
```suggestion
	const tosLink = 'https://automattic.com/for-agencies/platform-agreement/';
```

Comment (rcanepa) [reply client/a8c-for-agencies/sections/referrals/primary/footer/index.tsx]: I went ahead and added the link myself.

**Comment by jeffgolenski** [comment]
Notice is looking good, James. Just two small adjustments to help with legibility. Can we change the line-height and top margin of the description?

`.a4a-migration-offer__wrapper .a4a-migration-offer__description {
    color: var(--studio-gray-80);
    font-size: 1rem;
    font-weight: 400;
    line-height: 1.65;
    margin-top: 16px;`

(feel free to change the margin to `rem` is that's what we're working with)

Current:
CleanShot 2024-05-06 at 17 12 15


Proposed:
CleanShot 2024-05-06 at 17 13 44

File: client/a8c-for-agencies/sections/referrals/primary/footer/index.tsx (hunk 1 above)
Comment (jkguidaven) [reply client/a8c-for-agencies/sections/referrals/primary/footer/index.tsx]: Thanks for adding the link. 👍

**Comment by jkguidaven** [comment]
@jeffgolenski, thanks for the review and code snippet. I have applied it to the PR. 👍 

Screenshot 2024-05-07 at 6 41 59 PM

**Review by madebynoam — CHANGES_REQUESTED** [review]
This is looking good. Thanks @jkguidaven!

Just a few notes:
1. Can we add a line break on the title so that there is no widow, and reads a bit easier:
image

2. I saw that @jeffgolenski already covered the blue box edits.

**Comment by jkguidaven** [comment]
> This is looking good. Thanks @jkguidaven!
> 
> Just a few notes:
> 
> 1. Can we add a line break on the title so that there is no widow, and reads a bit easier:
> 
> image
> 2. I saw that @jeffgolenski already covered the blue box edits.

This has been addressed. 👍 

Screenshot 2024-05-07 at 8 41 41 PM

**Comment by madebynoam** [comment]
> > This is looking good. Thanks @jkguidaven!
> > Just a few notes:
> > 
> > 1. Can we add a line break on the title so that there is no widow, and reads a bit easier:
> > 
> > image
> > 2. I saw that @jeffgolenski already covered the blue box edits.
> 
> This has been addressed. 👍
> 
> Screenshot 2024-05-07 at 8 41 41 PM

Thank you, @jkguidaven!

**Review by madebynoam — APPROVED** [review]

**Pull request merged**